MigrationExecutorApp/ChangeFeedProcessortHost.cs

```diff
@@ -108,92 +121,131 @@ public async Task CloseAsync()
+                .WithInstanceName("consoleHost")
+                .WithLeaseContainer(sourceCollectionClient.GetContainer(config.LeaseDbName, config.LeaseCollectionName))
+                .WithStartTime(starttime)               
+                .WithMaxItems(1000)
+                .Build();
 
-            Trace.TraceInformation("Processor options Starts from Beginning - {0}, Lease renew interval - {1}",
-                processorOptions.StartFromBeginning,
-                processorOptions.LeaseRenewInterval.ToString());
+            await changeFeedProcessor.StartAsync().ConfigureAwait(false);
 
-            processorOptions.MaxItemCount = 1000;
-            var destClient = new DocumentClient(destCollInfo.Uri, destCollInfo.MasterKey,
-                new ConnectionPolicy() { ConnectionMode = ConnectionMode.Direct, ConnectionProtocol = Protocol.Tcp },
-                ConsistencyLevel.Eventual);
+            return changeFeedProcessor;
+        }
 
-            var docTransformer = new DefaultDocumentTransformer();
+        async Task ProcessChangesAsync(IReadOnlyCollection<Document> docs, CancellationToken cancellationToken)
+        {
+            Boolean isSyntheticKey = SourcePartitionKeys.Contains(",");
+            Boolean isNestedAttribute = SourcePartitionKeys.Contains("/");
+            Container targetContainer = destinationCollectionClient.GetContainer(config.DestDbName, config.DestCollectionName);
+            containerToStoreDocuments = targetContainer;
+            ConcurrentDictionary<int, int> failedMetrics = new ConcurrentDictionary<int, int>();
+            List<Task> tasks = new List<Task>();
+            List<Document> failedDocs = new List<Document>();
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            Document document = new Document();
+            foreach (Document doc in docs)
+            {
+                Console.WriteLine($"\tDetected operation...");
+                document = (SourcePartitionKeys != null & TargetPartitionKey != null) ? MapPartitionKey(doc, isSyntheticKey, TargetPartitionKey, isNestedAttribute, SourcePartitionKeys) : document = doc;
+                tasks.Add(containerToStoreDocuments.CreateItemAsync(item: document, cancellationToken: cancellationToken).ContinueWith((Task<ItemResponse<Document>> task) =>
+                {
+                    AggregateException innerExceptions = task.Exception.Flatten();
+                    CosmosException cosmosException = (CosmosException)innerExceptions.InnerExceptions.FirstOrDefault(innerEx => innerEx is CosmosException);
+                    failedMetrics.AddOrUpdate((int)cosmosException.StatusCode, 0, (key, value) => value + 1);
+                    WriteFailedDocsToBlob("FailedDocImport", containerClient, cosmosException, doc);
+                }, TaskContinuationOptions.OnlyOnFaulted));
+                // Simulate work
+                await Task.Delay(1);
```

Comment:
Is this delay needed?

Reply:
only to stop the "no await calls in method" warning, I guess

Reply:
There is an await in line 159, so I would remove it

Reply:
thanks - removed!